src/validateOptions.js

```diff
@@ -22,17 +19,29 @@ const ajv = new Ajv({
 errors(ajv);
 keywords(ajv, ['instanceof', 'typeof']);
 
-const validateOptions = (schema, options, name) => {
-  if (typeof schema === 'string') {
```

Comment:
I would recommend to move this breaking change into a separate PR. It make sense but it's not worth adding a new major version for only this change. All users would need to upgrade and not upgraded user won't benefit from the performance plus.

Reply:
makes sense